HLTrigger/Muon/plugins/HLTMuonTrkFilter.cc

```diff
@@ -96,11 +103,16 @@ bool HLTMuonTrkFilter::hltFilter(edm::Event& iEvent,
   std::vector<unsigned int> filteredMuons;
   for (unsigned int i = 0; i < muons->size(); ++i) {
     const reco::Muon& muon(muons->at(i));
+    const reco::RecoChargedCandidateRef cand(cands, i);
+    const reco::TrackRef tk = cand->track();
     // check for dR match to L1 muons
-    if (check_l1match) {
+    if (check_l1match && tk.isNonnull()) {
+      auto const propagated = prop.extrapolate(*tk);
+      auto const etaForMatch = propagated.isValid() ? propagated.globalPosition().eta() : cand->eta();
+      auto const phiForMatch = propagated.isValid() ? (double)propagated.globalPosition().phi() : cand->phi();
```

Comment:
```suggestion
    if (check_l1match) {
      const reco::RecoChargedCandidateRef cand(cands, i);
      const reco::TrackRef tk = cand->track();
      auto etaForMatch = cand->eta();
      auto phiForMatch = cand->phi();
      if (tk.isNonnull()) {
        auto const propagated = prop.extrapolate(*tk);
        if (propagated.isValid()) {
          etaForMatch = propagated.globalPosition().eta();
          phiForMatch = propagated.globalPosition().phi();
        }
      }
```
If `check_l1match` is `true`, but the trackRef is invalid, I would expect to have the old behaviour rather than skipping the matching condition altogether (due to `check_l1match && tk.isNonnull()` in the current impl). What is the difference between `TrackRef::isNonnull()` and `TrackRef::isValid()`? Should both be required?

Reply:
To answer the last one first, `TrackRef` does not have a `isValid()` function, so there is just `isNonnull()` to check. Otherwise, you are right about the logic and I implemented the suggestion.